tests/unit/client.py

```diff
@@ -0,0 +1,442 @@
+# Copyright 2013-2014 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
```

Comment:
So I **really** like some of the fixtures here, and I think there a great basis for writing some much needed tests of the server -- there's nowhere else where we're really testing the HTTP bits of the API; we're mostly calling the functions directly. However, it probably doesn't belong in the client library unit tests.

Furthermore, I don't think you need to run a real server process for these; you could just implement a mock HTTPClient that uses flask's test client.

I'd like to simplify this module a bit; it's a lot to digest, and much of it is inessential to unit testing the client library. That said, I *don't* want to just junk what you've written here, since it seems like a really good bit of integration testing for the server.